### PR TITLE
Adding functionality for the --strip-dot-git flag

### PR DESCRIPTION
**Pull request by rhwlo**
Mirroring https://github.com/rodjek/librarian-puppet/pull/332

r? @kitchen

### CODE REVIEW

File: features/install/git.feature [hunk 1]
```diff
@@ -197,3 +197,12 @@ Feature: cli/install/git
...
     And the output should contain "Dependency 'puppetlabs-stdlib' duplicated for module, merging"
+
+  Scenario: Omit the .git directory when passed --strip-dot-git
+    Given a file named "Puppetfile" with:
+    """
+    mod 'puppetlabs-stdlib',
+      :git => 'git://github.com/puppetlabs/puppetlabs-stdlib.git'
+    """
+    When I run `librarian-puppet install --strip-dot-git`
+    Then a directory "modules/stdlib/.git" should not exist
```
Comment (kitchen) [inline features/install/git.feature:208]: yay for ... cucumber?

(does this actually work? if so, awesome! :D)

Comment (rhwlo) [inline features/install/git.feature:208]: Yep! I added the test so that I could be sure :smile:

File: lib/librarian/puppet/source/git.rb [hunk 2]
```diff
@@ -37,6 +37,11 @@ def cache!
             raise Error, "Could not checkout #{uri}#{" at #{sha}" unless sha.nil?}: #{e}"
           end
 
+          if environment.config_db["install.strip-dot-git"] == "1"
+            dot_git = filesystem_path.join(".git")
+            dot_git.rmtree if dot_git.directory?
+          end
```
Comment (kitchen) [inline lib/librarian/puppet/source/git.rb:43]: question: if you, say, upgrade, does it actually work? I'm not certain how librarian-puppet knows what version exists in the modules directory without the .git directory, so I'd want to make sure that it's not re-cloning the upstream repo every time etc.

File: features/install/git.feature (hunk 1 above)
Comment (kitchen) [inline features/install/git.feature:208]: sweet! I haven't ever used cucumber before and the whole "write tests in english" felt very :wave: to me ;-)

**Comment by rhwlo** [comment]
This isn’t as simple as it initially seemed, and since I don’t have the time to tackle this at the moment, I’m going to close this PR.

**Pull request closed without merging**